modules/iasRtdProvider.js

```diff
@@ -51,94 +69,107 @@ function stringifyScreenSize() {
+function constructQueryString(anId, adUnits) {
   let queries = [];
   queries.push(['anId', anId]);
 
   queries = queries.concat(adUnits.reduce(function (acc, request) {
-    acc.push(['slot', stringifySlot(request, adUnitPath)]);
+    acc.push(['slot', stringifySlot(request)]);
     return acc;
   }, []));
 
   queries.push(['wr', stringifyWindowSize()]);
   queries.push(['sr', stringifyScreenSize()]);
   queries.push(['url', encodeURIComponent(window.location.href)]);
 
-  const queryString = encodeURI(queries.map(qs => qs.join('=')).join('&'));
-
-  const ajax = ajaxBuilder();
-
-  ajax(`${IAS_HOST}?${queryString}`, {
-    success: function (response, request) {
-      if (!isFinish) {
-        if (request.status === 200) {
-          const iasResponse = JSON.parse(response);
-          const commonBidResponse = {};
-          shallowMerge(commonBidResponse, getPageLevelKeywords(iasResponse));
-          commonBidResponse.slots = iasResponse.slots;
-          bidResponses = commonBidResponse;
-          adUnits.forEach(adUnit => {
-            adUnit.bids.forEach(bid => {
-              const rtd = bid.rtd || {};
-              const iasRtd = {};
-              iasRtd[SUBMODULE_NAME] = Object.assign({}, rtd[SUBMODULE_NAME], bidResponses);
-              bid.rtd = Object.assign({}, rtd, iasRtd);
-            });
-          });
-        }
-        isFinish = true;
-      }
-      callback();
-    },
-    error: function () {
-      utils.logError('failed to retrieve targeting information');
-      callback();
-    }
-  });
+  return encodeURI(queries.map(qs => qs.join('=')).join('&'));
+}
+
+function parseResponse(result) {
+  let iasResponse = {};
+  try {
+    iasResponse = JSON.parse(result);
+  } catch (err) {
+    utils.logError('error', err);
+  }
+  iasTargeting = iasResponse;
 }
 
 function getTargetingData(adUnits, config, userConsent) {
   const targeting = {};
-  Object.keys(bidResponses).forEach(key => bidResponses[key] === undefined ? delete bidResponses[key] : {});
   try {
-    adUnits.forEach(function(adUnit) {
-      targeting[adUnit] = bidResponses;
+    adUnits.forEach(function (adUnit) {
+      targeting[adUnit] = formatTargetingData(adUnit);
     });
   } catch (err) {
     utils.logError('error', err);
   }
+  utils.logInfo('IAS targeting', targeting);
   return targeting;
 }
 
+export function getApiCallback() {
+  return {
+    success: function (response, req) {
+      if (req.status === 200) {
+        try {
+          parseResponse(response);
+        } catch (e) {
+          utils.logError('Unable to parse IAS response.', e);
+        }
+      }
+    },
+    error: function () {
+      utils.logError('failed to retrieve IAS data');
+    }
+  }
+}
+
+function getBidRequestData(reqBidsConfigObj, callback, config, userConsent) {
+  const adUnits = reqBidsConfigObj.adUnits || getGlobal().adUnits;
+  const { pubId } = config.params;
+  const anId = pubId;
+  const queryString = constructQueryString(anId, adUnits);
```

Comment:
What purpose does `anId` serve here? Since it isn't used anywhere else it's less built code on page to pass config.params.pubId directly to the first arg of `constructQueryString`. I would recommend you include a check in your init function to verify that pubId is included in your rtd module params as well.

Reply:
@ncolletti thanks I resolved all your comment when you have time can you take a look?

Reply:
@raguilar-ias One of the comments was not resolved, also am curious on your thoughts on the suggestion to remove `anId` variable above. Thanks!

Reply:
@ncolletti anId is a variable is not a constant for that reason I added to config params.

Reply:
@ncolletti I removed anId variable you were right it was not used anywhere.